extension/src/workspace/workspaceInit.ts

```diff
@@ -0,0 +1,175 @@
+import * as vscode from 'vscode';
+import { AutolispGoToProvider } from "./gotoProvider";
+import { ReadonlyDocument } from "../project/readOnlyDocument";
+import * as fs from 'fs';
+
+
+// This variable represents un-opened *.LSP documents that we need to collect information from
+export const workspaceDocuments: Map<string, ReadonlyDocument> = new Map();
+export const activeDocuments: Map<string, vscode.TextDocument> = new Map();
+const docSelector = [ 'autolisp', 'lisp'];
+// How bad is it for me to store this context? I saw some other extensions doing this and vscode.extensions.getExtension() doesn't create a vscode.ExtensionContext
```

Comment:
I suggest to create a global GotoContextManager class to hold all these global vars, including the above the `activeDocuments`

Reply:
I was pretty worried about push back on this level of document management, but if I just need to be more formally handled, then I can do that.